pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -17,16 +17,58 @@
 
 import uuid
 from collections.abc import Hashable
+from functools import lru_cache
 from typing import Iterable, Optional, Union
 
 import numpy as np
 
-import pennylane as qml
+from pennylane import QuantumFunctionError, capture
 from pennylane.drawer.tape_mpl import _add_operation_to_drawer
 from pennylane.measurements.mid_measure import MeasurementValue, MidMeasureMP, measure
+from pennylane.ops.op_math import Conditional, adjoint
+from pennylane.ops.qubit import RX, RY, H, PhaseShift, S
+from pennylane.queuing import QueuingManager
+from pennylane.transforms import transform
 from pennylane.wires import Wires
 
 
+@lru_cache
+def _create_parametrized_mid_measure_primitive():
+    """Create a primitive corresponding to an mid-circuit measurement type.
+
+    Called when using :func:`~pennylane.measure`.
+
+    Returns:
+        jax.core.Primitive: A new jax primitive corresponding to a mid-circuit
+        measurement.
+
+    """
+    # pylint: disable=import-outside-toplevel
+    import jax
+
+    from pennylane.capture.custom_primitives import NonInterpPrimitive
+
+    mid_measure_p = NonInterpPrimitive("p_measure")
+
+    @mid_measure_p.def_impl
+    def _(wires, angle=0, plane="ZX", reset=False, postselect=None):
+        return _measure_impl(
+            wires,
+            measurement_class=ParametricMidMeasureMP,
+            angle=angle,
+            plane=plane,
+            reset=reset,
+            postselect=postselect,
+        )
+
+    @mid_measure_p.def_abstract_eval
+    def _(*_, **__):
+        dtype = jax.numpy.int64 if jax.config.jax_enable_x64 else jax.numpy.int32
```

Comment:
Is it possible to use `jax.numpy.int8` here?

Reply:
I suppose since this is in the FTQC module and only needs to work with our pipeline, we could? I'm not sure if we will run into problems later, since I guess it's meant to match the config. 

The main `measure` does this, and I would hesitate to change that, but we could add a custom capture for our `measure_z` function and then avoid using the standard `measure` for our workloads.